src/main/java/org/prebid/server/bidder/zeroclickfraud/ZeroclickfraudBidder.java

```diff
@@ -0,0 +1,154 @@
+package org.prebid.server.bidder.zeroclickfraud;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.iab.openrtb.request.BidRequest;
+import com.iab.openrtb.request.Imp;
+import com.iab.openrtb.response.BidResponse;
+import com.iab.openrtb.response.SeatBid;
+import io.vertx.core.http.HttpMethod;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.prebid.server.bidder.Bidder;
+import org.prebid.server.bidder.model.BidderBid;
+import org.prebid.server.bidder.model.BidderError;
+import org.prebid.server.bidder.model.HttpCall;
+import org.prebid.server.bidder.model.HttpRequest;
+import org.prebid.server.bidder.model.Result;
+import org.prebid.server.exception.PreBidException;
+import org.prebid.server.json.DecodeException;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.proto.openrtb.ext.ExtPrebid;
+import org.prebid.server.proto.openrtb.ext.request.zeroclickfraud.ExtImpZeroclickfraud;
+import org.prebid.server.proto.openrtb.ext.response.BidType;
+import org.prebid.server.util.HttpUtil;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * ZeroclickfraudBidder {@link Bidder} implementation.
+ */
+public class ZeroclickfraudBidder implements Bidder<BidRequest> {
+
+    private static final TypeReference<ExtPrebid<?, ExtImpZeroclickfraud>> ZEROCLICKFRAUD_EXT_TYPE_REFERENCE =
+            new TypeReference<ExtPrebid<?, ExtImpZeroclickfraud>>() {
+            };
+
+    private final String endpointTemplate;
+    private final JacksonMapper mapper;
+
+    public ZeroclickfraudBidder(String endpointTemplate, JacksonMapper mapper) {
+        this.endpointTemplate = HttpUtil.validateUrl(Objects.requireNonNull(endpointTemplate));
```

Comment:
validate url will fail on the real endpoint ("http://{{Host}}/openrtb2?sid={{SourceId}}")